manager/integration/tests/test_settings.py

```diff
@@ -653,11 +656,78 @@ def test_setting_backing_image_auto_cleanup():  # NOQA
         2. The related backing image file is removed.
         3. The download state of other disks keep unchanged.
            All volumes still work fine.
-    8. Delete all volumes. Verify that all states in the backing image disk map
-       will become terminating first,
-       and all entries will be removed from the map later.
+    8. Delete all volumes. Verify that there will only remain 1 entry in the
+       backing image disk map
     9. Delete the backing image.
     """
+    # Step 1
+    update_setting(client,
+                   "backing-image-cleanup-wait-interval",
+                   BACKING_IMAGE_CLEANUP_WAIT_INTERVAL)
+
+    # Step 2
+    create_backing_image_with_matching_url(
+            client, BACKING_IMAGE_NAME, BACKING_IMAGE_QCOW2_URL)
+
+    # Step 3
+    volume_names = [
+        volume_name + "-1",
+        volume_name + "-2",
+        volume_name + "-3"
+    ]
+
+    for volume_name in volume_names:
+        volume = create_and_check_volume(
+                            client, volume_name, 3,
+                            str(BACKING_IMAGE_EXT4_SIZE),
+                            BACKING_IMAGE_NAME)
+
+    # Step 4
+    lht_host_id = get_self_host_id()
+    for volume_name in volume_names:
+        volume = client.by_id_volume(volume_name)
+        volume.attach(hostId=lht_host_id)
+        wait_for_volume_healthy(client, volume_name)
+        assert volume.backingImage == BACKING_IMAGE_NAME
+
+    backing_image = client.by_id_backing_image(BACKING_IMAGE_NAME)
+    assert len(backing_image.diskFileStatusMap) == 3
+    for disk_id, status in iter(backing_image.diskFileStatusMap.items()):
+        assert status.state == "ready"
+        assert disk_id != ''
+
+    backing_images_in_disk = os.listdir("/var/lib/longhorn/backing-images")
+    assert len(backing_images_in_disk) == 1
+
+    # Step 5
+    for volume_name in volume_names:
+        volume = client.by_id_volume(volume_name)
+        volume.updateReplicaCount(replicaCount=1)
```

Comment:
I feel this step is unnecessary? because what we want to verify is `if a backing image in one disk is not referenced by any replicas, then it should be cleaned up after the wait interval. 

Basically, it's not related to how many replicas we need to change to test this.

Reply:
If I understanding correctly, this step is for the next step `6. Remove all replicas in one disk.` to make sure on a node there will be no replica exist to observe backing image on that node will auto clean up

But actually the replicaCount can set to 2, thank you

Reply:
Your understanding is partially correct, but the implementation here is missing some steps about replica scheduling. 

Firstly, removing replicas from the node does not mean new replicas rebuilt will not happen in the same node unless that node is unscheduled. 

Also, in your step 5, when scaling down the number of replicas, actually some replicas will be removed from any nodes, then you do the step 6, so it will probably cause some new replicas rebuilding and they are possibly recreated in the same node.

So, what you need to do can be: (however, need your verification)
- set your testing node (`lht_host_id`) unschedulable (I assume there is only one disk on that node, but technically you should make a disk unschedulable)
- then remove replicas on that node/disk, all new replicas will be created in other nodes
- then step 7

or 
- the same as the above first step
- scale down the number of replicas
- then step 7

Reply:
Hi @innobead ,

I have changed step 5 to disable test node and enable `Replica Node Level Soft Anti-Affinity ` to make sure no new replica will generate at next few steps.

Reply:
Actually U don't need to enable soft node anti affinity, but just keep it as default value.

Reply:
Here, what I want to clarify is that decreasing the replica count of a volume won't lead to Longhorn actively removing the redundant replica. Then the replica removal below is just deleting the extra/redundant replica on the current node without rebuilding.

Disabling the scheduling of the current node also works but there will be rebuilding triggered. I am fine with either solution. 

Yes, you don't need to enable soft node anti-affinity.

Reply:
> Here, what I want to clarify is that decreasing the replica count of a volume won't lead to Longhorn actively removing the redundant replica. Then the replica removal below is just deleting the extra/redundant replica on the current node without rebuilding.
> 

Thanks for the clarification on this! Outside the test here, so retaining the redundant ones seems intended even the number decreased, but why? Any context?

Reply:
Interesting, the context is here https://github.com/longhorn/longhorn-manager/pull/672#issuecomment-686045343. 

@shuo-wu Do we have any doc for this behavior? If not, I feel users will not know the redundant healthy replicas keeping alive even if the number of replicas has been decreased.

Reply:
No... The link may be not the context. IIRC, this behavior exists since the alpha version.

Reply:
@cchien816 ☝️ see this clarification from Shuo.